scipy/stats/tests/test_multivariate.py

```diff
@@ -2892,3 +2893,184 @@ def test_random_state_property():
+
+        y = dirichlet_multinomial.logpmf(alpha, x)
+        z = dirichlet_multinomial.pmf(alpha, x)
+        y_1 = dirichlet_multinomial.logpmf(alpha_1, x_1)
+        y_2 = dirichlet_multinomial.logpmf(alpha_2, x_2)
+        z_1 = dirichlet_multinomial.pmf(alpha_1, x_1)
+        z_2 = dirichlet_multinomial.pmf(alpha_2, x_2)
+
+        assert_equal(y[0], y_1)
+        assert_equal(y[1], y_2)
+        assert_equal(z[0], z_1)
+        assert_equal(z[1], z_2)
+
+    def test_broadcasting_mean(self):
+        n_1 = 3
+        n_2 = 4
+        alpha_1 = np.array([3, 4])
+        alpha_2 = np.array([6, 7])
+
+        alpha = np.array([[3, 4],
+                          [6, 7]])
+
+        n = np.array([3, 4])
+
+        y = dirichlet_multinomial.mean(alpha, n)
+        y_1 = dirichlet_multinomial.mean(alpha_1, n_1)
+        y_2 = dirichlet_multinomial.mean(alpha_2, n_2)
+        assert_equal(y[0], y_1)
+        assert_equal(y[1], y_2)
+
+    def test_broadcasting_var(self):
+        alpha = np.array([[0.3, 2.1, 5.4], [0.3, 2.1, 5.4], [0.3, 2.1, 5.4]]).T
+        n = 3
+        expected_v = np.array([[0.136162, 0.136162, 0.136162],
+                               [0.724381, 0.724381, 0.724381],
+                               [0.784293, 0.784293, 0.784293]])
+
+        v = dirichlet_multinomial.var(alpha, n)
+        assert_array_almost_equal(expected_v, v)
+
+    def test_broadcasting_cov(self):
+        rng = np.random.default_rng(2846)
+        alpha = rng.uniform(1e-10, 100, 10)
+        n = np.array([3, 4])
+
+        c = dirichlet_multinomial.cov(alpha, n)
+        c_0 = dirichlet_multinomial.cov(alpha, np.array([n[0]]))
+        c_1 = dirichlet_multinomial.cov(alpha, np.array([n[1]]))
+        assert_array_equal(c[0], c_0)
+        assert_array_equal(c[1], c_1)
+
+    def test_broadcasting_negative_x(self):
+        x = np.array([[1, 2, 3], [1, -3, 3], [1, 2, 3]])
+        alpha = np.array([[3, 4, 5], [3, 4, 5], [3, 4, 5]])
+        text = "`x` must not contain a non-negative integer."
+        with assert_raises(ValueError, match = text):
+            dirichlet_multinomial.logpmf(alpha, x)
+
+    def test_broadcasting_float_x(self):
+        x = np.array([[1, 2, 3], [1, 0.3, 3], [1, 2, 3]])
+        alpha = np.array([[3, 4, 5], [3, 4, 5], [3, 4, 5]])
+        text = "`x` must only contain integers."
+        with assert_raises(ValueError, match = text):
+            dirichlet_multinomial.logpmf(alpha, x)
+
+    def test_lengths(self):
+        x = np.array([1, 2, 3, 4])
+        alpha = np.array([3, 4, 5])
+        assert_raises(ValueError, dirichlet_multinomial.logpmf, alpha, x)
+
+    def test_postive_semi_definite(self):
+        #Makes sure that the covariance matrix
+        #is positive-semidefinite.
+        n = np.random.randint(0, 30)
+        alpha = np.random.random(10) * 5
+        n = np.array([n])
+        c = dirichlet_multinomial.cov(alpha, n)
+        eig = np.linalg.eigvals(c)
+        assert np.less_equal(eig, 0).any()
```

Comment:
This looks like a check for negative-semidefiniteness. If the covariance matrix is positive semidefinite and has at least one nonzero eigenvalue, I believe this will fail.

Reply:
Fixed it. Sorry about pushing a failed test, I've been having trouble with ```runtests.py``` lately.